sender/snmp.go

```diff
@@ -60,52 +63,56 @@ func (x *SNMP) Send(c *skogul.Container) error {
 		return x.err
 	}
 
-	pdutypes := []gosnmp.SnmpPDU{}
-
-	for _, m := range c.Metrics {
-		for j, i := range m.Data {
-			var pdutype gosnmp.SnmpPDU
-
-			pduName := fmt.Sprintf("%s", x.Oidmap[j])
-
-			switch reflect.TypeOf(i).Kind() {
-			case reflect.String:
-				pdutype = gosnmp.SnmpPDU{
-					Value: i,
-					Name:  pduName,
-					Type:  gosnmp.OctetString,
-				}
-			case reflect.Bool:
-				pdutype = gosnmp.SnmpPDU{
-					Value: i,
-					Name:  pduName,
-					Type:  gosnmp.Boolean,
-				}
-			case reflect.Int:
-				pdutype = gosnmp.SnmpPDU{
-					Value: i,
-					Name:  pduName,
-					Type:  gosnmp.Integer,
-				}
-			default:
+	pdutypes := []gosnmp.SnmpPDU{
+		{
+			Value: ".1.3.6.1.4.1.12748.2023.0.888",
+			Name:  ".1.3.6.1.6.3.1.1.4.1.0",
+			Type:  gosnmp.ObjectIdentifier,
+		},
+	}
+
+	m := c.Metrics[0]
+
+	for j, i := range m.Data {
+		var pdutype gosnmp.SnmpPDU
+
+		pduName := fmt.Sprintf("%s", x.Oidmap[j])
+
+		switch reflect.TypeOf(i).Kind() {
```

Comment:
I didn't realize this the last time I approved this, but this really doesn't require reflection.

You can write (copied from marshal.go as an example):

```
        switch value := v.(type) {
        case float64:
                d.Duration = time.Duration(value)
                return nil
        case string:
                var err error
                d.Duration, err = time.ParseDuration(value)
                if err != nil {
                        return err
                }
                return nil
        default:
                return errors.New("invalid duration")
        }

```

Reflection is considerably slower, which probably isn't noticeable in an SNMP trap sender, but should, in general, be avoided in busy loops.